cpp/include/cuspatial_test/vector_factories.cuh

```diff
@@ -35,6 +35,21 @@ auto make_device_vector(std::initializer_list<T> inl)
   return rmm::device_vector<T>(inl.begin(), inl.end());
 }
 
+template <typename T>
+auto make_device_uvector(std::initializer_list<T> inl,
```

Comment:
I thought that `device_uvector` meant uninitialized. If so, `make_device_uvector` from an initialization list doesn't make sense to me as a name.

Reply:
I think it's fine. a device_uvector is a container that is never *default* initialized. But you can still put data in it. This factory function just creates a uvector and then copies data into it.  

Note this is only available in tests, it is not intended for use in the library or by users of the library.  I kind of agree that we could probably get away with just `make_device_vector` in tests...

Reply:
I think the main reason why I added this helper is because `intersection_result` consists of `device_uvector`. And I tried to keep things simple by constructing `device_uvector` in test and just copy/move it to the struct. Using `device_vector` will require some extra steps.

Reply:
I don't require a change, but was mostly referring to the fact that the function name doesn't compute well for me. Since `uvector` takes an initializer list and that users are going to understand that it takes an initializer list, I'm ok with the name.

Reply:
I believe in c++ `make_*` is a convention for factory functions of classes. In standard c++, they were really meant to be a hack to make classes before CTAD. In cuDF, there are examples of building columns from various input specs that's non-standard to `cudf::column` ctor. I'm kinda abusing that idea here so it makes sense that it feels off. I'll add a line of comment to remind for better naming here.